Sources/XiEditor/Search.swift

```diff
@@ -321,6 +321,10 @@ extension EditViewController {
             updateScrollPosition(previousOffset: findViewController.view.fittingSize.height)
         }
 
+        if findViewController.searchQueries.isEmpty {
+            findViewController.addSearchField(searchField: nil)
```

Comment:
okay, there's a bug here: `addSearchField` changes the first responder, even if the find view isn't currently visible. This means that if cmd+D happens when  there are no queries, the editor window loses focus and you have to manually click on the view to regain.

I think this points at some architectural problems; the focus change really shouldn't be a side effect of a function that might be called in different places; so maybe either `addSearchField` needs to be split up, or the focus change should be moved to somewhere else?

Reply:
Maybe simply an additional `makeFirstResponder` parameter is sufficient for `addSearchField`?

Reply:
You could also do a check and only call `makeFirstResponder` if the view controller is visible.

Long-term there's some code smell here related to how we track state; the current control flow is pretty tough to follow, and I think that's the underlying cause of our original bug.